docs/section-7/feedforward-control.md

```diff
@@ -0,0 +1,213 @@
+
+Now that you have set the parameters, you need to specify which motors receive voltage and how you will log the data for analysis. In this example we will be controlling a single motor. The `Mechanism` object accepts those two functions:
+```java
+new SysIdRoutine.Mechanism(this::driveMotor, this::logMotor, this)
+```
+
+Here are the two functions that drive and log the motor voltage:
+```java
+// Mutable holder for unit-safe voltage values, persisted to avoid reallocation.
+private final MutableMeasure<Voltage> voltage = mutable(Volts.of(0));
+
+public void driveMotor(Measure<Voltage> volts) {
+    motor.setVoltage(volts.in(Volts));
+}
+
+public void logMotor(SysIdRoutineLog log) {
+    log.motor("shooter-motor")
+        .voltage(voltage.mut_replace(
+            motor.get() * RobotController.getBatteryVoltage(), 
+            Volts
+        ));
+}
+```
+Note that [SysIdRoutineLog](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj/sysid/SysIdRoutineLog.html) has a handy `motor` method which returns a `SysIdRoutineLog.MotorLog` object used to log voltage, linear position, velocity, and more. [Take a look at all the properties the object logs](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj/sysid/SysIdRoutineLog.MotorLog.html).
+
+You may also notice that the only values that the logger logs are of instances that must be of `MutableMeasure<(insert measure)>`. This records the values along with its units. You can't just log a value.
+
+!!! note
+    Notice that you can write anything in the `driveMotor()` and `logMotor()` methods. You are not limited to only powering a single motor but can power an entire elevator, arm, etc. SysID also analyzes elevators and arms which calculate the \\(k_{g}\\) constant.
+
+After you set up the testing parameters and mechanism to test, the SysIdRoutine provides functions that return a command to run the test.
+
+```java
+public Command sysIdQuasistatic(SysIdRoutine.Direction direction) {
+  return routine.quasistatic(direction);
+}
+
+public Command sysIdDynamic(SysIdRoutine.Direction direction) {
+  return routine.dynamic(direction);
+}
+```
+
+Typically it is recommended to bind these commands to controller buttons or an autonomous routine. If using a controller, it is recommended to bind them like this:
+
+```java
+GenericHID controller = new GenericHID(0);
+Shooter shooter = new Shooter();
+
+new JoystickButton(controller, Button.kY.value)
+    .whileTrue(
+        shooter.sysIdQuasistatic(SysIdRoutine.Direction.kForward)
+    );
+```
+Now you are ready to enable and run the tests! Typically the longer you run them, the more data you get which will lead to more accurate calculations. However, keep in mind not to run too long for safety purposes. After all four tests have been run, use the [DataLogTool](https://docs.wpilib.org/en/stable/docs/software/telemetry/datalog-download.html) to get the files.
```

Comment:
Nobody cares about safety.
I'm serious, you really don't need to include that line - people know if their robot is going to break something.

Reply:
ok sure